### ISSUE TITLE
[Bug]: Unable to start receiver com.onesignal.notifications.receivers.NotificationDismissReceiver: java.lang.reflect.InvocationTargetException

### ISSUE THREAD

**Issue by ajaybirla-fabzen** [OneSignal/OneSignal-Unity-SDK#678]
### What happened?

The application crashes whenever it receives a notification when it is in the background.

### Steps to reproduce?

```Markdown
1. Launch the application.
2. Minimize the application.
3. Send a notification to the application.
4. Click on the notification.
5. Observe the application crash.
```

### What did you expect to happen?

I expect the app to not crash whenever it receives a notification in the background.

### Unity version

2021.3.17f1

### OneSignal Unity SDK version

5.0.5

### Platform

Android

### Relevant log output

```Shell
Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.onesignal.notifications.receivers.NotificationDismissReceiver: java.lang.reflect.InvocationTargetException
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:4903)
       at android.app.ActivityThread.-$$Nest$mhandleReceiver()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2420)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8757)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

Caused by java.lang.reflect.InvocationTargetException:
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:90)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:417)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:196)
       at com.onesignal.notifications.receivers.NotificationDismissReceiver.onReceive(NotificationDismissReceiver.kt:36)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:4894)
       at android.app.ActivityThread.-$$Nest$mhandleReceiver()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2420)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8757)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

Caused by java.lang.NullPointerException:
       at com.onesignal.core.internal.application.impl.ApplicationService.getAppContext(ApplicationService.kt:39)
       at com.onesignal.core.internal.preferences.impl.PreferencesService.getSharedPrefsByName(PreferencesService.kt:157)
       at com.onesignal.core.internal.preferences.impl.PreferencesService.get(PreferencesService.kt:62)
       at com.onesignal.core.internal.preferences.impl.PreferencesService.getString(PreferencesService.kt:36)
       at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:126)
       at com.onesignal.common.modeling.SimpleModelStore.<init>(SimpleModelStore.kt:25)
       at com.onesignal.core.internal.config.ConfigModelStore.<init>(ConfigModelStore.kt:8)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:90)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:417)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:196)
       at com.onesignal.notifications.receivers.NotificationDismissReceiver.onReceive(NotificationDismissReceiver.kt:36)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:4894)
       at android.app.ActivityThread.-$$Nest$mhandleReceiver()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2420)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8757)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jonhy751** [comment]
I'm having the same problem.

Trace problem
Exception java.lang.RuntimeException: Unable to start receiver com.onesignal.notifications.receivers.NotificationDismissReceiver: java.lang.reflect.InvocationTargetException
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:4905)
  at android.app.ActivityThread.-$$Nest$mhandleReceiver
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2498)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:230)
  at android.os.Looper.loop (Looper.java:319)
  at android.app.ActivityThread.main (ActivityThread.java:8893)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:608)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1103)
Caused by java.lang.reflect.InvocationTargetException:
  at java.lang.reflect.Constructor.newInstance0
  at java.lang.reflect.Constructor.newInstance (Constructor.java:343)
  at com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:89)
  at com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
  at com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
  at com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:516)
  at com.onesignal.OneSignal.initWithContext (OneSignal.kt:208)
  at com.onesignal.notifications.receivers.NotificationDismissReceiver.onReceive (NotificationDismissReceiver.kt:39)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:4896)
Caused by java.lang.NullPointerException:
  at com.onesignal.core.internal.application.impl.ApplicationService.getAppContext (ApplicationService.kt:39)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.getSharedPrefsByName (PreferencesService.kt:234)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.get (PreferencesService.kt:134)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.getString (PreferencesService.kt:42)
  at com.onesignal.common.modeling.ModelStore.load (ModelStore.kt:165)
  at com.onesignal.common.modeling.SimpleModelStore.<init> (SimpleModelStore.kt:23)
  at com.onesignal.core.internal.config.ConfigModelStore.<init> (ConfigModelStore.kt:8)

**Comment by nan-li** [comment]
Hi all,
Thanks for reporting, this should be fixed in the latest [Unity SDK 5.1.1 Release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.1.1). Feel free to tag me if you are still seeing this issue.
